content/projects/gsoc/2022/project-ideas/JCasC-drift-detector.adoc

```diff
@@ -11,18 +11,28 @@ skills:
 mentors:
 - "jmMeessen"
 links:
-   emailThread: https://community.jenkins.io/t/google-summer-of-code-gsoc-2022-call-for-organizers-mentors-and-project-ideas/1010/2
+   emailThread: https://community.jenkins.io/t/gsoc-2022-project-proposal-jenkins-configuration-as-code-jcasc-drift-detector/1325
 ---
 
-**To Be Defined.**
 
-// === Background
-// TBD
+=== Context 
+A well designed Configuration Process will only allow updating Jenkins via Configuration as Code. 
+It will thus periodically and "blindly" overwrite the controller to make sure that the configuration code is equivalent to the running one. 
+This is the recommended practice (best if the interval is kept very short).
 
-// === Quick Start
-// TBD
+But experience shows that compromises need to be made. 
+These exceptions range from interactively validating touchy configuration change on Production, to allowing a developer to experiment the best parameters for a feature, or to an emergency interactive configuration change.
+This is typically the case when trying to reach the ideal Configuration Process maturity level. 
 
-// === Skills to Study and Improve
-// * TBD
+Another use case is to be able to compare the configuration of two Jenkins controllers (ex: Test vs. Production) 
+
+=== Project proposal
+
+This project intents to create the tooling, similar to the terraform plan command, to:
+
+* Show what configuration elements would be created or changed if the supplied JCasC file was applied to the given Jenkins system.
+** Ability to configure the details displayed for sensitive elements (credentials, sensitive configuration, etc)
+** Ability to opt out above mentioned sensitive elements. These elements may be absent or encrypted in the supplied JCasC file.
```

Comment:
```suggestion
** Ability to opt out of updating or displaying these sensitive elements. These elements may be absent or encrypted in the supplied JCasC file.
```
I don't know if this is correct but I think we need something -- just opting out of these elements doesn't make sense.

Reply:
Your suggestion clarified what I had in mind. But more thought on this is required in the early stages of that proposed project.